crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/MockWebServerStubs.java

```diff
@@ -3,49 +3,78 @@
 import okhttp3.mockwebserver.Dispatcher;
 import okhttp3.mockwebserver.MockResponse;
 import okhttp3.mockwebserver.RecordedRequest;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 
-import java.io.IOException;
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+import java.util.List;
 
 public class MockWebServerStubs {
 
-    private static final String REQUEST_PATH_EFORM_STAGING_WITH_NO_MAAT_REF = "/initialise/6000308";
-    private static final String REQUEST_PATH_EFORM_STAGING_WITH_MAAT_REF = "/initialise/6000309";
-    private static final String REQUEST_PATH_EFORM_STAGING_HUB_USER = "/initialise/6000310";
-    private static final String REQUEST_PATH_EFORM_STAGING_4XX = "/initialise/403";
-    private static final String REQUEST_PATH_CRIME_APPLY_DATASTORE_200OK_6000308 = "/6000308";
-    private static final String REQUEST_PATH_CRIME_APPLY_DATASTORE_200OK_6000309 = "/6000309";
-    private static final String REQUEST_PATH_CRIME_APPLY_4XX = "/403";
-
     public static Dispatcher forDownstreamApiCalls() {
         return new Dispatcher() {
+            @NotNull
             @Override
-            public MockResponse dispatch(RecordedRequest recordedRequest) {
-                return switch (recordedRequest.getPath()) {
-                    case REQUEST_PATH_CRIME_APPLY_DATASTORE_200OK_6000308 -> getMockResponse("data/crimeapply/MaatApplication_6000308.json");
-                    case REQUEST_PATH_CRIME_APPLY_DATASTORE_200OK_6000309 -> getMockResponse("data/crimeapply/MaatApplication_6000309.json");
-                    case REQUEST_PATH_CRIME_APPLY_4XX, REQUEST_PATH_EFORM_STAGING_4XX -> getMockResponseFor403();
-                    case REQUEST_PATH_EFORM_STAGING_WITH_NO_MAAT_REF -> getMockResponse("data/eformstaging/record_with_no_maatref.json");
-                    case REQUEST_PATH_EFORM_STAGING_WITH_MAAT_REF -> getMockResponse("data/eformstaging/record_with_maatref.json");
-                    case REQUEST_PATH_EFORM_STAGING_HUB_USER -> getMockResponse("data/eformstaging/record_created_by_hub.json");
-                    default -> new MockResponse().setResponseCode(503);
-                };
+            public MockResponse dispatch(@NotNull RecordedRequest recordedRequest) {
+                String requestPath = recordedRequest.getPath();
+                List<RequestPathResponseMapping> matchingMappings = Arrays.stream(RequestPathResponseMapping.values())
+                        .filter(requestPathResponseMapping -> requestPathResponseMapping.matchesRequestPath(requestPath))
```

Comment:
This logic can be simplified by having a HashMap inside enum with key, value for response file path and status code. Also, Arrays.steam and filter can be avoided

The dispatcher will return RequestPathResponseMapping.getMockResponse(requestPath);

```
private static final Map<String, String> BY_RESPONSE_FILE_PATH = new HashMap<>();
        private static final Map<String, HttpStatus> BY_HTTP_STATUS_CODE = new HashMap<>();

        static {
            for (RequestPathResponseMapping e : values()) {
                BY_RESPONSE_FILE_PATH.put(e.requestPath, e.responseBodyFilePath);
                BY_HTTP_STATUS_CODE.put(e.requestPath, e.httpResponseStatus);
            }
        }

      static MockResponse getMockResponse(String requestPath) {
            MockResponse mockResponse = new MockResponse()
                    .addHeader("Content-Type", MediaType.APPLICATION_JSON)
                    .setResponseCode(BY_HTTP_STATUS_CODE.get(requestPath).value());

            String responseFilePath = BY_RESPONSE_FILE_PATH.get(requestPath);
            if (StringUtils.isNotBlank(responseFilePath)) {
                mockResponse.setBody(FileUtils.readFileToString(responseFilePath));
            }
            return mockResponse;
        }
```

Reply:
I'll take a look and update

Reply:
I thought it might be neater to keep a single map with both request path and the RequestPathResponseMapping.
I felt the responsibility/behaviour of constructing a MockResponse is best placed as close as possible to the data which is required, i.e. on the enum itself.
Let me know what you think and any additional feedback you have :)

```
    private static final Map<String, RequestPathResponseMapping> PATH_TO_RESPONSE_MAPPING_MAP =
            Arrays.stream(RequestPathResponseMapping.values())
                    .collect(Collectors.toMap(RequestPathResponseMapping::getRequestPath,
                            requestPathResponseMapping -> requestPathResponseMapping));

    public static Dispatcher forDownstreamApiCalls() {
        return new Dispatcher() {
            @NotNull
            @Override
            public MockResponse dispatch(@NotNull RecordedRequest recordedRequest) {
                String requestPath = recordedRequest.getPath();
                RequestPathResponseMapping responseMapping = PATH_TO_RESPONSE_MAPPING_MAP.get(requestPath);

                if (responseMapping == null) {
                    return new MockResponse().setResponseCode(HttpStatus.SERVICE_UNAVAILABLE.value());
                }
                return responseMapping.getMockResponse();
            }
```

with the getMockResponse defined in the enum:

```
        MockResponse getMockResponse() {
            MockResponse mockResponse = new MockResponse()
                    .addHeader("Content-Type", MediaType.APPLICATION_JSON)
                    .setResponseCode(httpResponseStatus.value());

            if (StringUtils.isNotBlank(responseBodyFilePath)) {
                mockResponse.setBody(FileUtils.readFileToString(responseBodyFilePath));
            }
            return mockResponse;
        }
        };
    }
```

Reply:
Have added the below to the enum:

`
DEFAULT_FALLBACK_FOR_NULL_PATH(null, null, HttpStatus.SERVICE_UNAVAILABLE);
`

And as a result have removed the if condition:

```
            public MockResponse dispatch(@NotNull RecordedRequest recordedRequest) {
                String requestPath = recordedRequest.getPath();
                RequestPathResponseMapping responseMapping = PATH_TO_RESPONSE_MAPPING_MAP.get(requestPath);
                return responseMapping.getMockResponse();
            }
```